packages/devtools/lib/src/timeline/timeline_model.dart

```diff
@@ -277,11 +276,7 @@ class TimelineEvent {
 
   TimeRange time = TimeRange();
 
-  TimelineEvent parent;
```

Comment:
Nit: tweak TreeNode to be TreeNode<T>

and then this class is
class TimelineEvent extends TreeNode<TimelineEvent> {

}
That would avoida lot of these casts extra casts.

Reply:
Due to the recursive nature of the TreeNode members/methods, I'm not sure this is as simple as it sounds. For example, using the generic type T and changing the types of the members as such has some issues.
```
class TreeNode<T> {
   T parent;
   List<T> children;
}
```

Now when we do something like 
```
for (T child in children) {
  print(child.depth);
}
```

we have an error. Child no longer has a property depth because it is of type T, not TreeNode.

Reply:
The trick is to write
```dart
class TreeNode<T extends TreeNode> {

}```
let me know how that works.